### ISSUE TITLE
Connection loss handling

### ISSUE THREAD

**Issue by Akkjon** [featherbear/presonus-studiolive-api#34]
Hi,

I wondered if the libary already has, or potentially can have a way to handle the loss of connection.
At the moment if the connection to the mixer is lost, following error is thrown and the application crashes.

```
Error: read ECONNRESET
    at TCP.onStreamRead (node:internal/stream_base_commons:217:20) {
  errno: -4077,
  code: 'ECONNRESET',
  syscall: 'read'
}
```

Is there a way to hook into the connection loss to handle it and potentially try to reconnect?

Thanks :)

**Comment by featherbear** [comment]
Hi, currently there is no functionality to handle connection losses.

There are two cases that I have in mind to look for - JM Unsubscribe, and a general connection loss - this one can likely be handled [here](https://github.com/featherbear/presonus-studiolive-api/blob/master/src/lib/Client.ts#L166-L168).

I will eventually get around to implementing it, but feel free to have a crack in the mean time!

**Comment by Akkjon** [comment]
Implemented in #35

**Comment by featherbear** [comment]
Implemented in https://github.com/featherbear/presonus-studiolive-api/pull/35
